neworigins_v2/world.cpp

```diff
@@ -417,7 +461,38 @@ void Game::CreateWorld()
 	SetupNames();
 
 	regions.CreateNexusLevel( 0, nx, ny, "nexus" );
-	regions.CreateSurfaceLevel( 1, xx, yy, 0 );
+	if (generator == 1) {
+		regions.CreateSurfaceLevel( 1, xx, yy, 0 );
+	}
+	else {
+		int continents = -1;
+		while (continents < 16 || continents > 100) {
```

Comment:
can we calculate this at random based on a map size and water %?

Reply:
Yes... but why? This is a tool to set the desired _initial_ granularity of the map. The resulting number of islands still will be random.